### PR TITLE
refactor(rpc): extract `admin` and `txpool` into their respective crate

### LINKED ISSUE #894: [Feature] Extract `admin` and `txpool` into their own namespace crates
### Component

rpc

### Describe the feature you would like

To move `admin` and `txpool` namespace from `rpc-types-eth` into `rpc-types-admin` and `rpc-types-txpool` respectively

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Closes #894 

## Solution

- Extracts `admin` and `txpool` into their respective crates

In the `alloy` metacrate:

- Mark non-standard RPC crates optional
- When enabling the `rpc-types-*` flag enable its respective `api` flag in the provider
- Define a `rpc-types-debug` flag, that whilst it does not have a rpc-types implementation, it does consistenly enable its respective `debug-api` flag. This is similar in behavior to the `rpc-types-json` flag

Minor breaking changes

## PR Checklist

- [ ] Added Tests
- [x] Added Documentation
- [x] Breaking changes

### CODE REVIEW

File: crates/rpc-types-txpool/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#![doc = include_str!("../README.md")]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/alloy-rs/core/main/assets/alloy.jpg",
+    html_favicon_url = "https://raw.githubusercontent.com/alloy-rs/core/main/assets/favicon.ico"
+)]
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+
+pub mod txpool;
```
Comment (DaniPopes) [inline crates/rpc-types-txpool/src/lib.rs:9]: mods should be private if you re export everything

**Review by DaniPopes — CHANGES_REQUESTED** [review]
needs root readme update

**Comment by zerosnacks** [comment]
> needs root readme update

done

File: crates/rpc-types-beacon/src/lib.rs [hunk 2]
```diff
@@ -2,12 +2,13 @@
 //!
 //! Provides all relevant types for the various RPC endpoints, grouped by namespace.
 
+#![doc = include_str!("../README.md")]
```
Comment (Evalir) [inline crates/rpc-types-beacon/src/lib.rs:5]: is this a drive-by fix? don't think we touched the beacon crate in this PR

File: crates/rpc-types-beacon/src/lib.rs [hunk 3]
```diff
@@ -2,12 +2,13 @@
 //!
 //! Provides all relevant types for the various RPC endpoints, grouped by namespace.
```
Comment (Evalir) [inline crates/rpc-types-beacon/src/lib.rs:3]: seems this should be updated to say this is beacon-types-specific

File: crates/rpc-types-beacon/src/lib.rs (hunk 2 above)
Comment (zerosnacks) [reply crates/rpc-types-beacon/src/lib.rs:1]: Drive-by fix for consistency, removed the comment block for now, updated the README

File: crates/rpc-types-beacon/src/lib.rs (hunk 3 above)
Comment (zerosnacks) [reply crates/rpc-types-beacon/src/lib.rs]: Updated in README

**Review by DaniPopes — APPROVED** [review]

**Review by Evalir — APPROVED** [review]
lgtm
